./validate.sh broken on Go 1.10

See #466.

The tests fail with the following errors:

```
Running: go vet -source pbs_light.go
Running: go vet -source pbs_light_test.go
# command-line-arguments
./pbs_light_test.go:38:11: undefined: cookieSyncRequest
./pbs_light_test.go:54:12: undefined: cookieSyncResponse
./pbs_light_test.go:75:11: undefined: cookieSyncRequest
./pbs_light_test.go:97:12: undefined: cookieSyncResponse
./pbs_light_test.go:129:12: undefined: cookieSyncResponse
./pbs_light_test.go:161:12: undefined: cookieSyncResponse
./pbs_light_test.go:185:11: undefined: cookieSyncDeps
./pbs_light_test.go:303:2: undefined: sortBidsAddKeywordsMobile
./pbs_light_test.go:513:2: undefined: cacheVideoOnly
./pbs_light_test.go:513:32: undefined: auctionDeps
./pbs_light_test.go:513:32: too many errors
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,6 +1,7 @@
 language: go
 go:
     - "1.9"
+    - "1.10"
 
 env:
     - DEP_VERSION="0.4.1"

--- a/validate.sh
+++ b/validate.sh
@@ -20,7 +20,6 @@ done
 die() { echo -e "$@" 1>&2 ; exit 1;  }
 
 # Build a list of all the top-level directories in the project.
-# GOGLOB="*.go" -- need to do pbs_light.go and pbs_light_test.go in one command line for go 1.10
 for DIRECTORY in */ ; do
   GOGLOB="$GOGLOB ${DIRECTORY%/}"
 done
@@ -58,7 +57,7 @@ if [ "$RACE" -ne "0" ]; then
 fi
 
 if $VET; then
-  # Fix for the go 1.10 vet bug
+  # Fix for the go 1.10 vet bug (https://github.com/w0rp/ale/issues/1358)
   COMMAND="go tool vet -source *.go"
   echo "Running: $COMMAND"
   `$COMMAND`

--- a/validate.sh
+++ b/validate.sh
@@ -20,7 +20,7 @@ done
 die() { echo -e "$@" 1>&2 ; exit 1;  }
 
 # Build a list of all the top-level directories in the project.
-GOGLOB="*.go"
+# GOGLOB="*.go" -- need to do pbs_light.go and pbs_light_test.go in one command line for go 1.10
 for DIRECTORY in */ ; do
   GOGLOB="$GOGLOB ${DIRECTORY%/}"
 done
@@ -58,6 +58,10 @@ if [ "$RACE" -ne "0" ]; then
 fi
 
 if $VET; then
+  # Fix for the go 1.10 vet bug
+  COMMAND="go tool vet -source *.go"
+  echo "Running: $COMMAND"
+  `$COMMAND`
   for SOURCE in $GOGLOB ; do
     # default call for wildcards and directories
     COMMAND="go tool vet -source $SOURCE"

